### ISSUE TITLE
Error while training 3D_U_Net_training-clear.ipynb 

### ISSUE THREAD

**Issue by clarenzheng** [WenChentao/3DeeCellTracker#11]
I'm trying to train 3D U-net, but am getting the following error message. Each of my images (training labeling, training image, validation labeling, validation image) has shape of (328, 680, 15). Please let me know if you have any clues, thanks!
[3d_unet_training_error.txt](https://github.com/WenChentao/3DeeCellTracker/files/11922650/3d_unet_training_error.txt)

**Comment by WenChentao** [comment]
Thank you for asking. May I ask which architecture of unet you are using? By default, it is unet_3a, which accept input of shape (160,160,16). So your image depth is too small in size, which may cause this error.
You could try unet_3b instead, which accept input of shape (96,96,8). 

Alternativelly, you can upgrade 3DeeCellTracker to the latest version and use the StarDist model for segmentation.

If you meet new problems, please let me know.

**Comment by clarenzheng** [comment]
Thanks for your reply. I will try unet_3b. 

I have a question about the StarDist model. To what extent was the architecture edited, if at all, from the original program as found in the authors' repo? Specifically, what kinds of optimizations were made: "as we have optimized the StarDist-based tracking programs for more convenient and quick cell tracking"?

Thank you!

**Comment by WenChentao** [comment]
There are three big changes:
1. Replacing UNet + watershed by StarDist, which does not need users to fine-tune the segmentation parameters of noise level or the minimum cell size.
2. Normalizing the cell coordinates before tracking, so that the default values of the tracking parameters beta and lambda work well in most cases.
3. I released the notebook for users training FFN model using their custom data.

**Comment by clarenzheng** [comment]
Hi, thank you for the answer. Sorry that I was unclear, I meant to ask what kinds of changes, if any, were made to the StarDist model from the model described in the paper by Weigert et al?

**Comment by WenChentao** [comment]
I see. I simply wrote a wrapper to utilize the StarDist package in my program, without altering the architecture of the original model.

I tweaked the prediction method slightly to generate the probability map that is necessary for my tracking program, which was calculated but not returned in their original method. However, I did not modify the core part of the StarDist package.

**Comment by clarenzheng** [comment]
I see. Thank you!

**Comment by clarenzheng** [comment]
By the way, increasing my image stack file shapes to 16 or higher worked for the U-Net. Thanks for your tips.
